# HG changeset patch
# User Michael Goffioul
# Date 1209066177 14400
#      Thu Apr 24 15:42:57 2008 -0400
# Node ID bc5c6999c6009cdc5151faef2eb2f48b1ae7dec7
# Parent  2d2a969c731ca25a53191f3d428ebe1c2e818b4f
lo-sysdep.cc (octave_popen2): don't set PIPE_NOWAIT for parentWrite

diff --git a/liboctave/ChangeLog b/liboctave/ChangeLog
--- a/liboctave/ChangeLog
+++ b/liboctave/ChangeLog
@@ -1,8 +1,12 @@
+2008-04-24  Michael Goffioul  <michael.goffioul@gmail.com>
+
+	* lo-sysdep.cc (octave_popen2): Don't set PIPE_NOWAIT for parentWrite.
+
 2008-04-21  John W. Eaton  <jwe@octave.org>
 
 	* idx-vector.cc (IDX_VEC_REP::idx_vector_rep (const boolNDArray&)):
 	Fix for-loop condition.
 
 2008-04-19  Jaroslav Hajek <highegg@gmail.com>
 
 	* CmplxCHOL.cc, CmplxQR.cc, dbleCHOL.cc, dbleQR.cc: Fix calls to error()
diff --git a/liboctave/lo-sysdep.cc b/liboctave/lo-sysdep.cc
--- a/liboctave/lo-sysdep.cc
+++ b/liboctave/lo-sysdep.cc
@@ -136,17 +136,16 @@ octave_popen2 (const std::string& cmd, c
     {
       msg = "popen2: pipe creation failed";
       return -1;
     }
   if (! sync_mode)
     {
       pipeMode = PIPE_NOWAIT;
       SetNamedPipeHandleState (parentRead, &pipeMode, 0, 0);
-      SetNamedPipeHandleState (parentWrite, &pipeMode, 0, 0);
     }
   fildes[1] = _open_osfhandle (reinterpret_cast<long> (parentRead), _O_RDONLY | _O_BINARY);
   fildes[0] = _open_osfhandle (reinterpret_cast<long> (parentWrite), _O_WRONLY | _O_BINARY);
   si.dwFlags |= STARTF_USESTDHANDLES;
   si.hStdInput = childRead;
   si.hStdOutput = childWrite;
 
   // Ignore first arg as it is the command
